substratevm/mx.substratevm/mx_substratevm.py

```diff
@@ -176,6 +180,11 @@ def _run_graalvm_cmd(cmd_args, config, nonZeroIsFatal=True, out=None, err=None,
         primary_suite_dir = config.primary_suite_dir
     else:
         config_args = []
+        if not _jlink_libraries():
+            config_args += ['--no-jlinking']
+        # 'libpoly' is special, we need to exclude it instead of including
```

Comment:
@olpaw I am not sure this is the right thing to do here, but that's what I understood based on other parts. The Polyglot Library will essentially get automatically added (if necessary), while explicitly listing it under `--components` results in:
```
Exception: Unknown component: Polyglot Library
```

Reply:
You can avoid thinking about `libpoly` handling by instead using `mx_sdk_vm_impl._components_include_list()`
```patch
Index: substratevm/mx.substratevm/mx_substratevm.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/substratevm/mx.substratevm/mx_substratevm.py b/substratevm/mx.substratevm/mx_substratevm.py
--- a/substratevm/mx.substratevm/mx_substratevm.py	(revision Staged)
+++ b/substratevm/mx.substratevm/mx_substratevm.py	(date 1632819141389)
@@ -180,11 +177,10 @@
         primary_suite_dir = config.primary_suite_dir
     else:
         config_args = []
-        if not _jlink_libraries():
+        if not mx_sdk_vm_impl._jlink_libraries():
             config_args += ['--no-jlinking']
         # 'libpoly' is special, we need to exclude it instead of including
-        components = [c for c in mx_sdk_vm_impl.registered_graalvm_components() if c.short_name != 'libpoly']
-        config_args += ['--components=' + ','.join([x.name for x in components])]
+        config_args += ['--components=' + ','.join(c.name for c in mx_sdk_vm_impl._components_include_list())]
         dynamic_imports = [x for x, _ in mx.get_dynamic_imports()]
         if dynamic_imports:
             config_args += ['--dynamicimports', ','.join(dynamic_imports)]

```
This way `mx.run_mx` runs with the exact same copy of `--dynamicimports` that you used in the top-level mx call.